fix var args for shape guards

avikchaudhuri opened the pull request:
Summary: Fixes #162599

Test Plan:
added test based on repro

Rollback Plan:

Differential Revision: D82144520

tugsbayasgalan commented on test/export/test_export.py:
```diff
@@ -4826,6 +4826,51 @@ def forward(self, x, repeat):
+
+        w = Wrapped()
+
+        if is_retracebility_test(self._testMethodName):
```
Why is this ok to fail?

avikchaudhuri replied:
This is probably some existing issue with how dynamic shapes are specified with var args.

avikchaudhuri commented on torch/export/_unlift.py:
```diff
@@ -82,6 +82,41 @@ def _check_inputs_match(args, kwargs, in_spec: pytree.TreeSpec) -> list:
+        # match substrings of the form L['<name>'][<number>]
+        orig_arg_names.update(re.findall(r'L\[\'([^\']+)\'\]\[([0-9]+)\]', g))
+
+    sig_arg_names = set()
```
I'll wait for you to land internally.

avikchaudhuri committed:
fix var args for shape guards (#162633)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/162633

Fixes #162599

Test Plan:
added test based on repro

Rollback Plan:

Reviewed By: tugsbayasgalan

Differential Revision: D82144520
